### PR TITLE
Symlink premium on atomic and expand the use of the premium modal

### PR DESCRIPTION
**Pull request by brezocordero**
## Description

This PR replaces the download and activate premium links on the key verification screens and on the upgrade premium banner with the modal to install and activate the premium plugin

It also changes the installation of the premium plugin to use the symlinked version if the plugin is in the Dotcom platform

## Code review notes

_N/A_

## QA notes

To test:
- Get a key that gives premium features
- Install the plugin on a new site
- Go through the wizard, and check that the key verification works as expected
- Delete the premium plugin
- Go to Settings > Key activation. Replace the key with an invalid key and click on verify.
- Reload the page
- Add the valid key, click on verify
- Check that you see the button to download the plugin
- Click on the button, check that you see the modal with a button to download and not to purchase
- Click on download, check that the modal changes to Activate
- Click on activate, reload the page
- Check that premium is installed and active
- Deactivate or delete the plugin
- Go to Automations > click on one of the premium features, check that the modal still works

To test on Dotcom:
- Unzip the build, change the name of the plugin folder and zip it back (this is required or the plugin will be replaced with the .org version)
- Install the modified zip on a Dotcom business site (make sure it is using the build and not the managed version)
- Use a key with premium features and follow any path that will show the modal to download the premium plugin
- Click on download
- Click on activate
- Go to plugins, check that MailPoet premium is active and managed by Host

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5435]

## After-merge notes

_N/A_

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [x] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-5435]: https://mailpoet.atlassian.net/browse/MAILPOET-5435?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/lib/API/JSON/v1/Premium.php [hunk 1]
```diff
@@ -7,11 +7,13 @@
 use MailPoet\Config\AccessControl;
 use MailPoet\Config\ServicesChecker;
 use MailPoet\WP\Functions as WPFunctions;
+use MailPoet\WPCOM\DotcomHelperFunctions;
 use WP_Error;
 
 class Premium extends APIEndpoint {
   const PREMIUM_PLUGIN_SLUG = 'mailpoet-premium';
   const PREMIUM_PLUGIN_PATH = 'mailpoet-premium/mailpoet-premium.php';
+  const SIMLINK_PATH = '../../../../wordpress/plugins/mailpoet-premium/latest';
```
Comment (costasovo) [inline mailpoet/lib/API/JSON/v1/Premium.php]: I assume this is where the plugin always is on the dotcom. I would add this info to the constant name and maybe add a little comment that the path is relative to WP_PLUGIN_DIR. When I looked for the first time, I thought it was relative to this file.

**Review by costasovo — COMMENTED** [review]
Thank you for improving the premium plugin installation experience! 
The code looks good! I added just one comment with a small naming suggestion.

File: mailpoet/lib/API/JSON/v1/Premium.php (hunk 1 above)
Comment (brezocordero) [reply mailpoet/lib/API/JSON/v1/Premium.php]: Done!

**Review by costasovo — DISMISSED** [review]
Thanks for improving the constant name!

**Comment by Aschepikov** [comment]
HI @brezocordero there are some conflicts can you resolve them pls

**Review by Aschepikov — DISMISSED** [review]
GJ!

**Comment by Aschepikov** [comment]
HI @brezocordero there are some conflicts can you resolve them pls

**Comment by brezocordero** [comment]
@costasovo Would you mind taking a quick look? 
I have rebased and there was a recent change in the upgrade banner. This PR replaces it with the modal. 
I spoke with @NeosinneR and said it was ok to replace it as we should use the modal whenever possible.
I believe it is the same behavior but would like a quick validation.
Thanks!

**Review by costasovo — DISMISSED** [review]
LGTM! Thank you for improving the API key activation experience. I'm not aware of any issue with the behavior.

**Comment by Aschepikov** [comment]
Hi, @brezocordero Also I have this link to download the premium plugin, should it be covered by a symlink?
image

**Comment by NeosinneR** [comment]
> I have this link to download the premium plugin, should it be covered by a symlink?

@brezocordero @Aschepikov yeah, this would be also good to cover, so the experience of installing the premium plugin is unified.

**Comment by brezocordero** [comment]
@Aschepikov I spoke with @NeosinneR and the PHP notice will be covered on a different ticket.

I rebased again trunk and it requires a new validation.
Would you mind doing one last test with the upgrade banner that appears on the segments editing page?

Thank you so much.

**Comment by Aschepikov** [comment]
Hi @brezocordero  I have a few issues with this pr
1. I cant activate the premium via the activate button
image
Tried on 2 sites from automations and segments, 
2. Invalid API endpoint. error when I am trying to open the email statistics page
image
Checked on 2 sites too
3. On the subscriber's statistics page I have the upgrade premium banner in cases when it should be the download and the activate.
image

**Comment by brezocordero** [comment]
@Aschepikov Sorry for that!

It is a version issue with the premium plugin. My build of the free plugin is more advanced than the version of the Premium plugin you get when you install it via the modal.
If you test on a Jurassic ninja, you get a fatal with this version of the free plugin and current version of the Premium plugin.

To verify it is that, you can check the debug.log file and see if you have an error about 
```
PHP Fatal error:  Declaration of MailPoet\Premium\Automation\Integrations\MailPoetPremium\Actions\UnsubscribeAction::run(MailPoet\Automation\Engine\Data\StepRunArgs $args): void must be compatible with MailPoet\Automation\Engine\Integration\Action::run(MailPoet\Automation\Engine\Data\StepRunArgs $args, MailPoet\Automation\Engine\Control\StepRunController $controller): void in /srv/users/userfaf548a5/apps/userfaf548a5/public/wp-content/plugins/mailpoet-premium/lib/Automation/Integrations/MailPoetPremium/Actions/UnsubscribeAction.php on line 65
```
You would need to test the activation with the latest build of premium.

**Review by costasovo — APPROVED** [review]
I didn't notice changes except for rebasing. So I'm approving.

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**